### PR TITLE
Refactor component creation script

### LINKED ISSUE #336: Refactor component script to use async
Avoids promise callback chaining and reduces `*Sync` methods in fs.

### PR DESCRIPTION
**Pull request by kmonahan**
Closes #336 . Reworks parts of the component generator script to use async/await instead of fs.*Sync methods and promise chaining, as was originally suggested in #334 . No changes to the functionality from an end-user perspective.

### CODE REVIEW

File: lib/component.js
Comment (gustavderdrache) [inline lib/component.js]: If you're just returning a promise, you don't need to `await` it here (under the hood, it forces two yields: the first is the `await` inside this function, followed by any callers' `await`).

File: lib/component.js
Comment (gustavderdrache) [inline lib/component.js]: Let's clean this up to use `path.join` instead of concatenation.

File: lib/component.js
Comment (gustavderdrache) [inline lib/component.js]: I'd probably rename variables so that you're able to use destructuring:

```js
const prompt = [...];
const { confirm } = await inquirer.prompt(prompt);
```

File: lib/component.js
Comment (gustavderdrache) [inline lib/component.js]: The promise returned by `makeComponentFile` isn't being used by this arrow function, so it's probably simple to remove the `async`/`await` here (same reason as mentioned in an earlier comment - the extra `await` forces an unnecessary yield).

File: lib/component.js
Comment (gustavderdrache) [inline lib/component.js]: Why not `await mkdirp(location)`? It'll save you some nesting.

File: lib/component.js
Comment (gustavderdrache) [inline lib/component.js]: Remember to `.catch` any async errors thrown by this function. Node will track rejected promises, but it's fairly inconsistent - right now it's a warning, but eventually it'll be a hard error.

**Review by gustavderdrache — APPROVED** [review]
It looks like there are a few superfluous `await`s that can be removed, and I've left comments on a few minor style nits, but there's really nothing major as far as I'm concerned.

Overall, a very nice cleanup!

**Comment by kmonahan** [comment]
Thanks, @gustavderdrache ! @coreylafferty, fyi, I'd like to make the changes suggested here before this one gets merged.

**Comment by kmonahan** [comment]
Changes made! This is ready to merge (unless anyone else has feedback).

**Pull request merged**
